blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -12,33 +12,31 @@ import scalaz.{-\/, \/-}
 object BlazeClient {
   def apply(manager: ConnectionManager, idleTimeout: Duration, requestTimeout: Duration): Client = {
     Client(Service.lift { req =>
-      def tryClient(client: BlazeClientStage, freshClient: Boolean, flushPrelude: Boolean): Task[DisposableResponse] = {
+      val key = RequestKey.fromRequest(req)
+      def tryClient(client: BlazeClientStage, flushPrelude: Boolean): Task[DisposableResponse] = {
         // Add the timeout stage to the pipeline
         val ts = new ClientTimeoutStage(idleTimeout, requestTimeout, bits.ClientTickWheel)
         client.spliceBefore(ts)
         ts.initialize()
 
         client.runRequest(req, flushPrelude).attempt.flatMap {
-          case \/-(r)    =>
+          case \/-(r)  =>
             val dispose = Task.delay {
-              if (!client.isClosed()) {
-                ts.removeStage
```

Comment:
This is the reason timeouts are causing a problem: the timeout stage is no longer being removed.